reference-implementation/lib/helpers.js

```diff
@@ -62,6 +62,10 @@ exports.IsFiniteNonNegativeNumber = v => {
 };
 
 exports.InvokeOrNoop = (O, P, args) => {
+  assert(exports.typeIsObject(O) === true);
```

Comment:
Ah, right, I didn't notice that distinction.

So we have to allow `underlyingSource` et al. to be a boolean, string, or symbol, and treat it like they passed `{}` i.e. no methods?